Microsoft/WebTemplateStudio
# Create resource group for Azure CosmosDB

Related PBI: https://microsoftgarage.visualstudio.com/Intern%20GitHub/_workitems/edit/30315

The purpose of this PR is to create resource group for user if they choose to put the Azure Functions and/or CosmosDB they create in a new resource group.

Client side changes will be in another PR (the client will be sending an empty string for `functions.resourceGroup` and/or `cosmos.resourceGroup` field in the payload when user choose a new resource for the respective services). 

So, to test this PR:
- Add **_(1)_** `payload.functions.resourceGroup = "";` and/or ### _(2)_ `payload.cosmos.resourceGroup = "";` before line 92 in `generationExperience.ts`. This will force the method that deploys creation of resource group to be kicked off. 
- Launch the wizard
- Take note of the project name you entered in the first page of the wizard. 
- Add Azure Functions and/or CosmosDB when creating your project
- Note: **if you added _(1)_**, the resource group you select in the Azure Functions form will not matter, because it will be deployed to the new resource group, but please check to see if your CosmosDB is deployed to the selected resource group, same goes for **_(2)_**.
- Note: **if you added _(1)_ and _(2)_**, the resource group you select in the Azure Functions form and CosmosDB form will not matter, because both of them will be deployed to the new resource group.
- Go onto Azure portal and check if a new resource group has been created. The name of the new resource group should start with the project name your entered
- Make sure that the services you chose are deployed to the right resource groups

**Note**: The arm template required for resource group in the generated project will also be addressed in another PR.
